### DISCUSSION TITLE
Google Maps integration

### DISCUSSION THREAD

**Discussion by krishna-tech-dot** [clientIO/joint#2764] [Q&A]
### Introduction

Hi Joint js team,

Is there any example where joint js+ integration with Google Maps React(react-google-maps).
(https://developers.google.com/maps/documentation/javascript).
(https://visgl.github.io/react-google-maps/examples).

if possible, please share the link of the example or implementation of the Joint Js+ integration with Google Maps.

Thanks in advance.

Regards,
Krishna Chaitanya

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Here is an example of how to integrate **Google Maps** and **JointJS**.

https://codepen.io/jointjs/pen/QWeyYza??editors=0010

_Note: I experimented a while ago, but never finished it. However, you might find it useful._

**Reply by krishna-tech-dot** [reply]
Hi @kumilingus,

Tried implementing the above example using '**@vis.gl/react-google-maps**',

Facing issue while creating the CustomOverlay for JointJs+.

I cannot use the overlay directly from google.maps, OverlayView is part of the react hook - useMapsLibrary (like MapsLibrary -> OverlayView).

Can you please help me out with any react example using the above library.

**Reply by kumilingus** [reply]
Sorry, I am not planning to dedicate any more time to this (at least not at this moment).

If the React hook doesn't expose everything you need, you may want to contact the developers of the Google Maps React wrapper.
